commons/src/main/java/com/navercorp/pinpoint/common/util/NetUtils.java

```diff
@@ -36,32 +36,77 @@
 
     public static final String LOOPBACK_ADDRESS_V4 = "127.0.0.1";
 
+
+    private static final HostAndPortFactory<InetSocketAddress> inetSocketAddressFactory = new HostAndPortFactory<InetSocketAddress>() {
+        @Override
+        public InetSocketAddress newInstance(String host, int port) {
+            if (!HostAndPort.isValidPort(port)) {
+                return null;
+            }
+            return new InetSocketAddress(host, port);
+        }
+    };
+
     private NetUtils() {
     }
 
     public static List<InetSocketAddress> toInetSocketAddressLIst(List<String> addressList) {
-        List<InetSocketAddress> inetSocketAddressList = new ArrayList<InetSocketAddress>();
+        return toHostAndPortLIst(addressList, inetSocketAddressFactory);
+    }
 
+
+    public interface HostAndPortFactory<T> {
+        T newInstance(String host, int port);
+    }
+
+    public static <T> List<T> toHostAndPortLIst(List<String> addressList, HostAndPortFactory<T> hostAndPortFactory) {
+        if (CollectionUtils.isEmpty(addressList)) {
+            return Collections.emptyList();
+        }
+        final List<T> hostAndPortList = new ArrayList<T>(addressList.size());
         for (String address : addressList) {
-            InetSocketAddress inetSocketAddress = toInetSocketAddress(address);
-            if (inetSocketAddress != null) {
-                inetSocketAddressList.add(inetSocketAddress);
+            final T hostAndPort = parseHostAndPort(address, hostAndPortFactory);
+            if (hostAndPort != null) {
+                hostAndPortList.add(hostAndPort);
             }
         }
-
-        return inetSocketAddressList;
+        return hostAndPortList;
     }
 
+
     public static InetSocketAddress toInetSocketAddress(String address) {
-        try {
-            URI uri = new URI("pinpoint://" + address);
+        return parseHostAndPort(address, inetSocketAddressFactory);
+    }
 
-            return new InetSocketAddress(uri.getHost(), uri.getPort());
-        } catch (URISyntaxException ignore) {
-            // skip
+    public  static <T> T parseHostAndPort(String address, HostAndPortFactory<T> hostAndPortFactory) {
+        if (StringUtils.isEmpty(address)) {
+            return null;
         }
+        Assert.requireNonNull(address, "address must not be null");
```

Comment:
It looks like duplicate validation.
If it is not a duplicate validation, then order should change.
or It might be a validation for hostAndPortFactory variable.

Reply:
opps! 👍